database-commons/src/main/java/io/cdap/plugin/db/batch/config/DatabaseSinkConfig.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.batch.config;
+
+import java.util.List;
+
+/**
+ * Interface for DB Sink plugin config
+ */
+public interface DatabaseSinkConfig extends DatabaseConnectionConfig {
+
+  /**
+   * @return the reference name of the sink stage
+   */
+  String getReferenceName();
+
+  /**
+   * @return the transaction isolation level
+   */
+  String getTransactionIsolationLevel();
+
+  /**
+   * @return the initial query to be run upon connecting to database
+   */
+  List<String> getInitQueries();
+
+  /**
+   * @return true if any of the connection parameters is macro, otherwise false
+   */
+  boolean connectionParamsContainsMacro();
```

Comment:
I think it is fine to just rename all places with same name to `canConnect`, don't think it will result in any incompatibility.